oci/table_ oci_kms_vault.go

```diff
@@ -0,0 +1,250 @@
+package oci
+
+import (
+	"context"
+	"strings"
+
+	oci_common "github.com/oracle/oci-go-sdk/v36/common"
+	"github.com/oracle/oci-go-sdk/v36/keymanagement"
+	"github.com/turbot/go-kit/types"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableKmsVault(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_kms_vault",
+		Description: "OCI KMS Vault",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AnyColumn([]string{"id"}),
+			Hydrate:    getKmsVault,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listKmsVaults,
+		},
+		GetMatrixItem: BuildCompartementRegionList,
+		Columns: []*plugin.Column{
+			{
+				Name:        "display_name",
+				Description: "A user-friendly name. Does not have to be unique, and it's changeable.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "The OCID of a vault.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel(),
+			},
+			{
+				Name:        "lifecycle_state",
+				Description: "A vault's current lifecycle state.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "time_created",
+				Description: "The date and time a vault was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
+			},
+			{
+				Name:        "crypto_endpoint",
+				Description: "The service endpoint to perform cryptographic operations against.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "management_endpoint",
+				Description: "The service endpoint to perform management operations against.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "vault_type",
```

Comment:
There are many properties like `WrappingkeyId` , `TimeOfDeletion`, `RestoredFromVaultId`  which are we get in `get` call is missing, please check add the appropriate columns.